packages/block-editor/src/components/iframe/get-compatibility-styles.js

```diff
@@ -0,0 +1,120 @@
+let compatibilityStyles = null;
+
+/**
+ * Returns a list of stylesheets that target the editor canvas. A stylesheet is
+ * considered targetting the editor a canvas if it contains the
+ * `editor-styles-wrapper`, `wp-block`, or `wp-block-*` class selectors.
+ *
+ * Ideally, this hook should be removed in the future and styles should be added
+ * explicitly as editor styles.
+ */
+export function getCompatibilityStyles() {
+	if ( compatibilityStyles ) {
+		return compatibilityStyles;
+	}
+
+	// Only memoize the result once on load, since these stylesheets should not
+	// change.
+	compatibilityStyles = Array.from( document.styleSheets ).reduce(
+		( accumulator, styleSheet ) => {
+			try {
+				// May fail for external styles.
+				// eslint-disable-next-line no-unused-expressions
+				styleSheet.cssRules;
+			} catch ( e ) {
+				return accumulator;
+			}
+
+			const { ownerNode, cssRules } = styleSheet;
+
+			// Stylesheet is added by another stylesheet. See
+			// https://developer.mozilla.org/en-US/docs/Web/API/StyleSheet/ownerNode#notes.
+			if ( ownerNode === null ) {
+				return accumulator;
+			}
+
+			if ( ! cssRules ) {
+				return accumulator;
+			}
+
+			// Don't try to add the reset styles, which were removed as a dependency
+			// from `edit-blocks` for the iframe since we don't need to reset admin
+			// styles.
+			if ( ownerNode.id === 'wp-reset-editor-styles-css' ) {
+				return accumulator;
+			}
+
+			// Don't try to add styles without ID. Styles enqueued via the WP dependency system will always have IDs.
+			if ( ! ownerNode.id ) {
+				return accumulator;
+			}
+
+			function matchFromRules( _cssRules ) {
+				return Array.from( _cssRules ).find(
+					( {
+						selectorText,
```

Comment:
@ellatrix On WP.com, I'm seeing an error being thrown on this line:
```
Error: Permission denied to access property "selectorText"
```

<img width="1163" alt="Screenshot 2024-01-22 at 10 17 06" src="https://github.com/WordPress/gutenberg/assets/664258/e7dfee8f-58b9-4422-9a74-3c856e9c8454">

Any idea what could it be? It happens in Firefox, Chrome seems to be fine.

Reply:
This didn't happen previously? I didn't really change anything here, I just moved the code to a new file.

Reply:
I don't know, it just started happening randomly on a draft post. And after closing and opening the same post in a new tab the error disappeared again, so I don't really have any steps to reproduce.

But I'm wondering what "Permission denied to access property" even means. It's like the stylesheet object was from a cross-origin iframe, and therefore behind some security boundary.

Reply:
That's why there's a try/catch for accessing `styleSheet.cssRules`, normally it fails there for cross origin stylesheets